### PR TITLE
Add API StorageDocumentProvider.setDocumentContent(IDocument, String)

### PR DESCRIPTION
**Pull request by jukzi**
* use IFile.readString()
* deprecate for removal: setDocumentContent(document, contentStream)

### CODE REVIEW

File: bundles/org.eclipse.ui.editors/src/org/eclipse/ui/editors/text/StorageDocumentProvider.java [hunk 1]
```diff
@@ -136,14 +136,25 @@ protected void setDocumentContent(IDocument document, InputStream contentStream,
...
+			setDocumentContent(document, content);
 		} catch (IOException x) {
 			String message= (x.getMessage() != null ? x.getMessage() : ""); //$NON-NLS-1$
 			IStatus s= new Status(IStatus.ERROR, PlatformUI.PLUGIN_ID, IStatus.OK, message, x);
 			throw new CoreException(s);
 		}
 	}
 
+	/**
+	 * Initializes the given document with the given stream using the given encoding.
```
Comment (iloveeclipse) [inline bundles/org.eclipse.ui.editors/src/org/eclipse/ui/editors/text/StorageDocumentProvider.java]: Javadoc copy/pasted?

Comment (jukzi) [reply bundles/org.eclipse.ui.editors/src/org/eclipse/ui/editors/text/StorageDocumentProvider.java]: yes, my bad. thanks!

File: bundles/org.eclipse.ui.editors/src/org/eclipse/ui/editors/text/FileDocumentProvider.java [hunk 2]
```diff
@@ -371,38 +368,7 @@ protected boolean setDocumentContent(IDocument document, IEditorInput editorInpu
 	protected boolean setDocumentContent(IDocument document, IEditorInput editorInput, String encoding) throws CoreException {
```
Comment (jukzi) [inline bundles/org.eclipse.ui.editors/src/org/eclipse/ui/editors/text/FileDocumentProvider.java:368]: parameter "encoding" is not used

**Review by jukzi — COMMENTED** [review]
-1

**Pull request closed without merging**